mindsdb_server/namespaces/datasource.py

```diff
@@ -208,97 +308,147 @@ class Analyze(Resource):
     def get(self, name):
         global global_mdb
         ds = get_datasource(name)
-        if ds['name'] is None:
+        if ds is None:
             print('No valid datasource given')
             abort(400, 'No valid datasource given')
 
         if 'analysis_data' in ds and ds['analysis_data'] is not None:
             return ds['analysis_data'], 200
 
-        analysis = global_mdb.analyse_dataset(ds['source'], sample_margin_of_error=0.025)
+        analysis = get_analysis(ds['source'])
 
         ds['analysis_data'] = analysis
         save_datasource_metadata(ds)
         
         return analysis, 200
 
 
+@ns_conf.route('/<name>/analyze_subset')
+@ns_conf.param('name', 'Datasource name')
+class AnalyzeSubset(Resource):
+    @ns_conf.doc('analyse_datasubset')
+    def get(self, name):
+        global global_mdb
+        ds = get_datasource(name)
+        if ds is None:
+            print('No valid datasource given')
+            abort(400, 'No valid datasource given')
+
+        ds_dir = os.path.join(mindsdb.CONFIG.MINDSDB_DATASOURCES_PATH, ds['name'], 'datasource')
+        db_path = os.path.join(ds_dir, 'sqlite.db')
+
+        where = []
+        for key, value in request.args.items():
+            if key.startswith('filter'):
+                param = parse_filter(key, value)
+                if param is None:
+                    abort(400, f'Not valid filter "{key}"')
+                where.append(param)
+
+        con = sqlite3.connect(db_path)
+        cur = con.cursor()
+
+        column_names = get_sqlite_columns_names(cur)
+        where = [x for x in where if x['field'] in column_names]
+        where, marks = prepare_sql_where(where)
+
+        query = ' '.join(['select * from data', where])
+        cur.execute(query, marks)
+        data = cur.fetchall()
+        data = [dict(zip(column_names, x)) for x in data]
+
+        cur.close()
+        con.close()
+
+        if len(data) == 0:
+            return abort(400, 'Empty dataset after filters applying')
+
+        temp_file_fd, temp_file_path = tempfile.mkstemp(prefix='mindsdb_data_subset_', suffix='.csv', dir='/tmp')
+        with open(temp_file_path, 'w') as csvfile:
+            writer = csv.DictWriter(csvfile, fieldnames=column_names)
+            writer.writeheader()
+            for row in data:
+                writer.writerow(row)
+
+        analysis = get_analysis(temp_file_path)
```

Comment:
This seems like an approach both tedious and prone to namgin conflicst.

Why not do:

`analysis = get_analysis(pd.DataFrame(data))`

instead ?

Reply:
I have tried that first before chose way with temp file. But analyse_dataset from dataFrame raise error, so i thought it not implemented.
```

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:52 - [START] DataExtractor

Exception: Could not load module StatsGenerator
INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:57 - [END] DataExtractor, execution time: 0.040 seconds

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:52 - [START] DataCleaner

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:57 - [END] DataCleaner, execution time: 0.027 seconds

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:52 - [START] StatsGenerator

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/stats_generator/data_preparation.py:17 - Analyzing a sample of 2 from a total population of 2, this is equivalent to 100.0% of your data.

ERROR:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/controllers/transaction.py:130 - Could not load module StatsGenerator

ERROR:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/controllers/transaction.py:131 - Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/mindsdb/libs/controllers/transaction.py", line 127, in _call_phase_module
    return module(self.session, self)(**kwargs)
  File "/usr/local/lib/python3.6/dist-packages/mindsdb/libs/phases/base_module.py", line 54, in __call__
    ret = self.run(**kwargs)
  File "/usr/local/lib/python3.6/dist-packages/mindsdb/libs/phases/stats_generator/stats_generator.py", line 557, in run
    col_data = clean_int_and_date_data(col_data, self.log)
  File "/usr/local/lib/python3.6/dist-packages/mindsdb/libs/phases/stats_generator/data_preparation.py", line 26, in clean_int_and_date_data
    if str(ele) not in ['', str(None), str(False), str(np.nan), 'NaN', 'nan', 'NA', 'null'] and (not ele or not ele.isspace()):
AttributeError: 'int' object has no attribute 'isspace'

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:52 - [START] DataExtractor

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:57 - [END] DataExtractor, execution time: 0.142 seconds

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:52 - [START] DataCleaner

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:57 - [END] DataCleaner, execution time: 0.025 seconds

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/base_module.py:52 - [START] StatsGenerator

INFO:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/phases/stats_generator/data_preparation.py:17 - Analyzing a sample of 2 from a total population of 2, this is equivalent to 100.0% of your data.

ERROR:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/controllers/transaction.py:130 - Could not load module StatsGenerator

ERROR:mindsdb-logger-3b706f34-a5a6-11ea-836d-2c56dc4ecd27:libs/controllers/transaction.py:131 - Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/mindsdb/libs/controllers/transaction.py", line 127, in _call_phase_module
    return module(self.session, self)(**kwargs)
  File "/usr/local/lib/python3.6/dist-packages/mindsdb/libs/phases/base_module.py", line 54, in __call__
    ret = self.run(**kwargs)
  File "/usr/local/lib/python3.6/dist-packages/mindsdb/libs/phases/stats_generator/stats_generator.py", line 557, in run
    col_data = clean_int_and_date_data(col_data, self.log)
  File "/usr/local/lib/python3.6/dist-packages/mindsdb/libs/phases/stats_generator/data_preparation.py", line 26, in clean_int_and_date_data
    if str(ele) not in ['', str(None), str(False), str(np.nan), 'NaN', 'nan', 'NA', 'null'] and (not ele or not ele.isspace()):
AttributeError: 'int' object has no attribute 'isspace'
```